### ISSUE TITLE
Training crash at first epoch

### ISSUE THREAD

**Issue by AC82-AI** [Danand/audio-sample-generator#3]
Now I'm trying to start the train of my single files but as soon as it starts, at first epoch, it crash python.
Here's the log:
/Users/me/Documents/GitHub/audio-sample-generator/audio-sample-generator/.venv/lib/python3.10/site-packages/torch/utils/checkpoint.py:61: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/Users/me/Documents/GitHub/audio-sample-generator/audio-sample-generator/.venv/lib/python3.10/site-packages/torch/utils/checkpoint.py:429: UserWarning: torch.utils.checkpoint: please pass in use_reentrant=True or use_reentrant=False explicitly. The default value of use_reentrant will be updated to be False in the future. To maintain current behavior, pass use_reentrant=True. It is recommended that you use use_reentrant=False. Refer to docs for more details on the differences between the two variants.
  warnings.warn(
/AppleInternal/Library/BuildRoots/d9889869-120b-11ee-b796-7a03568b17ac/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:725: failed assertion `[MPSNDArray initWithDevice:descriptor:] Error: total bytes of NDArray > 2**32'
./run.sh: line 21: 44712 Abort trap: 6           streamlit run Getting_Started.py
mycomputer:audio-sample-generator acik$ /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 3 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Comment by rei-koshka** [comment]
@AC82-AI 
Hi!  
I've encountered this warning too, but it didn't caused training failure.

You could try to pull latest commit and run the app with fresh environment:
```bash
git pull
rm -rf .venv
./run.sh
```

**Comment by AC82-AI** [comment]
Hey Hi @Danand thanks for you're reply.
Tried what you suggest but seems it does not work. 
Could be probably that I'm on macOS and it's trying to use MPS instead of CUDA?
Of course I've tried "export PYTORCH_ENABLE_MPS_FALLBACK=1" before running the env, but no other results than crash.

**Comment by rei-koshka** [comment]
>Could be probably that I'm on macOS and it's trying to use MPS instead of CUDA?

@AC82-AI 
Yes, probably. I thought about it. But the script `run.sh` already has `export PYTORCH_ENABLE_MPS_FALLBACK=1` 
To be honest, I didn't check if the app successfully runs on CUDA. But CPU and MPS worked well.

Have you tried running training with fewer images?

**Comment by AC82-AI** [comment]
Sure I've tried with just one since I've started checking it. Then if everything worked well on CPU or MPS on your side then there's something else that I can't understand.
